tests/collections/collection.test.ts

```diff
@@ -2173,7 +2174,7 @@ describe(`StargateMongoose - ${testClientName} Connection - collections.collecti
         { sort: { username: 1 } }
       );
 
-      const docs = await collection.find({}, { sort: { username: 1 } }).toArray();
+      const docs = await collection.find({}, { sort: { username: 1 }, limit: 20 }).toArray();
```

Comment:
Can we add a couple of test cases for negative and zero limits as well and have the behavior of JSON API tested?

Reply:
@kathirsvn done in 873d6da . Turns out we treat limit 0 as no limit, which is consistent with how MongoDB and Mongoose handle it.

Negative limits throw an error in stargate-mongoose, which is slightly different than MongoDB. [MongoDB does support negative limits](https://www.mongodb.com/docs/manual/reference/method/cursor.limit/#negative-values); for practical purposes, `limit: -2` is equivalent to `limit: 2`, there's just slightly different behavior in the case where `limit > batchSize`. However, negative limits in MongoDB are rarely used and I don't think we should prioritize supporting negative limits.